File: assignment 6/c/programs/1.c
#include <stdio.h>
#define MAX_SIZE 100

int stack[MAX_SIZE];
int top = -1;

void push(int element)
{
    if (top == MAX_SIZE - 1)
    {
        printf("Stack overflow\n");
        return;
    }
    stack[++top] = element;
}

int pop()
{
    if (top == -1)
    {
        printf("Stack underflow\n");
        return -1;
    }
    return stack[top--];
}

int main()
{
    push(10);
    push(20);
    push(30);
    printf("%d\n", pop());
    printf("%d\n", pop());
    printf("%d\n", pop());
    printf("%d\n", pop());
    return 0;
}

File: assignment 6/python/programs/1.py
class Stack:
    def __init__(self):
        self.stack = []

    # Use list append method to push element in stack
    def push(self, item):
        self.stack.append(item)

    # Use list pop method to pop element from stack
    def pop(self):
        if len(self.stack) < 1:
            return None
        return self.stack.pop()

    # Use peek to look at the top of the stack without removing an element
    def peek(self):
        if not self.stack:
            return None
        return self.stack[-1]

    # Check if the stack is empty
    def is_empty(self):
        return len(self.stack) == 0

# Test the Stack class
s = Stack()
s.push("A")
s.push("B")
s.push("C")
print(s.pop())
print(s.peek())
